examples/src/main/scala/io/finch/todo/Main.scala

```diff
@@ -31,21 +31,23 @@ object Main extends TwitterServer {
 
   private val port: Flag[Int] = flag("port", 8081, "TCP port for HTTP server")
 
-  def main(): Unit = {
-    println(s"Open your browser at http://localhost:${port()}/todo/index.html") // scalastyle:ignore
-
-    val server = for {
-      id <- Ref[IO].of(0)
-      store <- Ref[IO].of(Map.empty[Int, Todo])
-    } yield {
-      val app = new App(id, store)(IO.contextShift(ExecutionContext.global))
-      val srv = Http.server.withStatsReceiver(statsReceiver)
-
-      srv.serve(s":${port()}", app.toService)
+  private def serve(service: Service[Request, Response]): Resource[IO, ListeningServer] =
+    Resource.make(for {
+      srv <- IO(Http.server.withStatsReceiver(statsReceiver))
+      server <- IO(srv.serve(s":${port()}", service))
+    } yield server) { server =>
+      IO.defer(ToAsync[Future, IO].apply(server.close()))
     }
 
-    val handle = server.unsafeRunSync()
-    onExit(handle.close())
-    Await.ready(adminHttpServer)
-  }
+  val run: IO[Unit] =
+    (for {
+      _ <- Resource.eval(IO.println(s"Open your browser at http://localhost:${port()}/todo/index.html")) // scalastyle:ignore
+      id <- Resource.eval(Ref[IO].of(0))
+      store <- Resource.eval(Ref[IO].of(Map.empty[Int, Todo]))
+      app = new App(id, store)
+      service <- app.toService
+      server <- serve(service)
+    } yield server).use(_ => IO(Await.ready(adminHttpServer)) >> IO.never)
```

Comment:
Should we drop IO.never here? I think it we just keep waiting on `Await.ready`, the app would correctly shutdown when /quitquitqui endpoint is hit.